frontend/src/components/movies/MovieDetails.js

```diff
@@ -8,271 +8,222 @@ import { popupNotification } from '../../lib/notification'
 
 import { FaHeart } from 'react-icons/fa'
 
-class MovieDetails extends React.Component {
-  state = {
-    movie: null,
-    formdata: {
-      text: '',
-      rating: ''
-    },
-    liked_movies: '',
-    currentUserId: '',
-    liked_by: '', 
-    heartColor: 'grey',
-    liked: false,
-    commentErrors: {}
-  }
-  
+const MovieDetails = (props) => {
+  const { match } = props
 
-  
-  async componentDidMount() {    
-    // request single movie by id
-    const movieId = this.props.match.params.id
-    // console.log(this.props, movieId)
-    const response = await getSingleMovie(movieId)
-    // console.log(response.data)
-    this.setState({
-      movie: response.data
-    })  
+  const [movie, setMovie] = useState('')
+  const [formData, setFormdata] = useState({ text: '', rating: '' })
+  const [currentUserId, setCurrentUserId] = useState('')
```

Comment:
Current users should be controlled by a context that wraps this component instead of storing these types of unrelated variables inside the component itself.